### PR TITLE
C3PO Updates

### PR DESCRIPTION
**Pull request by ejm201**
Summary of Changes:

9/8/2021

Campaign Exclusions

Added /campaign_exclusions route that allows for the exclusion of transaction and player ids on specified campaigns.

Campaign Export

Added /export_demographics route that allows for an CSV export of campaign information.

Campaigns

Added ability to include any question_id from the /qualifications route as a substitution value to the respondents demographics portion of supplier link attribute.

Added /redirect_test route that retrieves the redirect_test_url for a specific campaign.

Custom Screeners

We now have the ability to create/edit/destroy custom screeners (qualifications) and as a result, the following routes have been added.

/custom_qualifications
/custom_qualification_answers
/campaign_qualifications

Campaign Quotas
Added disclaimer on usage of multiple location qualifications on a single quota.

### CODE REVIEW

**Review by chris-tap — APPROVED** [review]

File: source/includes/_custom_screeners.md [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+question_text | String | This is the question text that will be served to our respondents.
+answer_type | Integer | This value will determine how campaign_qualifications pre-codes are generated for quotas. See **Answer Types** section for more information.
+country_language_id | Integer | Country language targeting for the qualification.
+
+### Optional Parameters
+Parameter | Type | Description
+--------- | ---- | -----------
+name | String | Name of the custom qualification.
+country | String | Country that is being targeted.
+country_id | Integer | Country specific ID used for internal purposes.
```
Comment (itsmikeq) [inline source/includes/_custom_screeners.md]: It does not look like the endpoint takes a `country` or `country_id`, rather just the `country_language_id`

File: source/includes/_custom_screeners.md [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+customer_id | Integer | The unique identifier associated with your account. See **Callbacks** section for more information.
```
Comment (itsmikeq) [inline source/includes/_custom_screeners.md]: This is set automatically and users are unable to change this parameter

File: source/includes/_custom_screeners.md [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+
+### Response
+Parameter | Type | Description
+--------- | ---- | -----------
+name | String | Name of the custom qualification.
+question_id | Integer | This value will be used to fetch qualification details.
+question_text | String | This is the question text that will be served to our respondents.
+answer_type | Integer | This value will determine how campaign_qualifications pre-codes are generated for quotas. See **Answer Types** section for more information.
```
Comment (itsmikeq) [inline source/includes/_custom_screeners.md]: I think this is referring to checkbox, multiple select, etc from [here](https://github.com/TapResearch/raynor/blob/master/config/initializers/active_enum.rb#L202) but i may be incorrect.

File: source/includes/_custom_screeners.md (hunk 1 above)
Comment (ejm201) [reply source/includes/_custom_screeners.md]: I saw some reference in the spec but this makes sense I took these two out.

File: source/includes/_custom_screeners.md [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+## Create a campaign qualification
+
+Create a campaign qualification which will allow you to associate your custom qualifications and answers to a specific campaign.
+
+> Sample Request Payload - 
+
+```json
+{
+    "question_id": 100112713,
+    "pre_code_values": "1"
```
Comment (itsmikeq) [inline source/includes/_custom_screeners.md:111]: This is referred to as an Int in line 100 but then is a string here.  Lets use string everywhere for pre_code_values

File: source/includes/_custom_screeners.md [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+id | Integer | This the unique identifier of the campaign qualification.You will need to use this parameter to update a specific campaign qualification.
+question_id | Integer | This value will be used to fetch qualification details.
+pre_code_values | String | The pre_code value received in the response of returned when creating a custom qualification answer. See **Create a custom qualification answer** for more information.
+
+
+### Response
+Parameter | Type | Description
+--------- | ---- | -----------
+question_id | Integer | This value will be used to fetch qualification details.
+pre_codes | String | Values to be passed in the pre_codes array to describe which respondents qualify for a survey.
```
Comment (itsmikeq) [inline source/includes/_custom_screeners.md]: It looks like the response here is an array of `int`'s, right? 
https://github.com/TapResearch/slate/pull/5/files#diff-c9d1c5474bc24df142fcac5764bab9930c429f049932b93ad34d0e9c805e1287R123

File: source/includes/_custom_screeners.md (hunk 4 above)
Comment (ejm201) [reply source/includes/_custom_screeners.md:111]: All references to pre_code_values have been changed to string.

File: source/includes/_custom_screeners.md (hunk 5 above)
Comment (ejm201) [reply source/includes/_custom_screeners.md]: Fixed!

File: source/includes/_custom_screeners.md (hunk 3 above)
Comment (ejm201) [reply source/includes/_custom_screeners.md]: Updated the copy and link to this newly written article: **https://app.intercom.com/a/apps/uaut5pir/articles/articles/5569336/show**

File: source/includes/_custom_screeners.md (hunk 2 above)
Comment (ejm201) [reply source/includes/_custom_screeners.md]: Updated!

**Review by itsmikeq — APPROVED** [review]
LGTM!

**Pull request merged**